### PR TITLE
Handling file types via mime types (fixes PDF issues)

### LINKED ISSUE #1268: Handling file types via mime types (fixes PDF issues)

### PR DESCRIPTION
**Pull request by Omri-Levy**

### CODE REVIEW

File: apps/backoffice-v2/src/common/components/organisms/ImageViewer/ImageViewer.ZoomModal.tsx [hunk 1]
```diff
@@ -32,28 +32,30 @@ export const ZoomModal: FunctionComponent<IZoomModalProps> = ({
       hideTitle
       {...rest}
     >
-      {(selectedImage?.fileType === 'pdf' || selectedImage?.imageUrl?.endsWith('.pdf')) && (
+      {(selectedImage?.fileType === 'application/pdf' ||
```
Comment (Blokh) [inline apps/backoffice-v2/src/common/components/organisms/ImageViewer/ImageViewer.ZoomModal.tsx]: extract to method

File: apps/backoffice-v2/src/common/components/organisms/ImageViewer/ImageViewer.SelectedImage.tsx [hunk 2]
```diff
@@ -28,7 +28,10 @@ export const SelectedImage = forwardRef<HTMLImageElement | HTMLIFrameElement, TS
       setIsError(false);
     }, [isLoading, selectedImage?.imageUrl]);
 
-    if (selectedImage?.fileType === 'pdf' || selectedImage?.imageUrl?.endsWith('.pdf')) {
+    if (
+      selectedImage?.fileType === 'application/pdf' ||
```
Comment (Blokh) [inline apps/backoffice-v2/src/common/components/organisms/ImageViewer/ImageViewer.SelectedImage.tsx]: use memtype as the "new" way to descover file type.
and keep old one for backword compatibility until we migrate all the relevant information of files.
add TODO so we wont forget to remove it later on.

File: apps/backoffice-v2/src/pages/Entity/hooks/useEntity/mock-workflow-with-children.ts [hunk 3]
```diff
@@ -671,7 +671,7 @@ export const workflow = {
                   url: 'https://alchemy.veriff.com/v/eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2ODk4NjExNDAsInNlc3Npb25faWQiOiJlZWIyOTY3Yy0xMGI2LTRhZDktYjllNy1kZGNlNjg4NjcyOTYiLCJpaWQiOiI0NGFhNDc1Ny02NzdhLTQ3ZWUtYTE3Ny0zZjA2MmY1NjI0OTkifQ.bAq1VFSgNbps1bu1dEouz79JXQUSY9Gn38SchgmVYzs',
                 },
               },
-              vendor: 'veriff',
+              vendor: 'test',
```
Comment (Blokh) [inline apps/backoffice-v2/src/pages/Entity/hooks/useEntity/mock-workflow-with-children.ts]: update back to veriff

File: packages/common/src/schemas/documents/default-context-schema.json [hunk 4]
```diff
@@ -149,7 +149,7 @@
                   "format": "uri"
                 },
                 "type": {
-                  "enum": ["pdf", "png", "jpg"]
+                  "enum": ["application/pdf", "image/png", "image/jpg"]
```
Comment (Blokh) [inline packages/common/src/schemas/documents/default-context-schema.json]: revert

File: services/workflows-service/scripts/workflows/kyc-dynamic-process-example.ts [hunk 5]
```diff
@@ -88,7 +88,7 @@ export const kycDynamicExample = {
               address: {
                 fullAddress: entity.data.additionalInfo.address
                 },
-              vendor: 'veriff'
+              vendor: 'test'
```
Comment (Blokh) [inline services/workflows-service/scripts/workflows/kyc-dynamic-process-example.ts]: revert

**Review by Blokh — APPROVED** [review]

File: services/workflows-service/prisma/migrations/20230903084419_renamed_file_type_column/migration.sql [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/*
+  Warnings:
+
+  - You are about to drop the column `type` on the `File` table. All the data in the column will be lost.
+
+*/
+-- AlterTable
+ALTER TABLE "File" DROP COLUMN "type",
```
Comment (MatanYadaev) [inline services/workflows-service/prisma/migrations/20230903084419_renamed_file_type_column/migration.sql:8]: Why did you drop and recreate `type`?

File: services/workflows-service/prisma/migrations/20230830103231_file_type_field/migration.sql [hunk 7]
```diff
@@ -0,0 +1,2 @@
+-- AlterTable
+ALTER TABLE "File" ADD COLUMN     "type" TEXT;
```
Comment (MatanYadaev) [inline services/workflows-service/prisma/migrations/20230830103231_file_type_field/migration.sql:2]: Redundant migration file. Ideally you should remove it and just fix your local DB.

File: services/workflows-service/prisma/migrations/20230903084419_renamed_file_type_column/migration.sql [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+ADD COLUMN     "mimeType" TEXT;
```
Comment (MatanYadaev) [inline services/workflows-service/prisma/migrations/20230903084419_renamed_file_type_column/migration.sql:9]:
```suggestion
ALTER TABLE "File" ADD COLUMN     "mimeType" TEXT;
```

File: services/workflows-service/scripts/workflows/parent-kyb-workflow.ts [hunk 9]
```diff
@@ -78,7 +78,7 @@ export const kybParentDynamicExample = {
                 countryOfIncorporation: entity.data.countryOfIncorporation,
                 companyNumber: entity.data.registrationNumber,
                 state: entity.data.additionalInfo.company.state
-                vendor: 'open-corporates'
+                vendor: 'veriff'
```
Comment (MatanYadaev) [inline services/workflows-service/scripts/workflows/parent-kyb-workflow.ts:81]: Why `veriff` instead of `open-corporates`?

**Review by MatanYadaev — APPROVED** [review]

**Pull request merged**